[c10d] Change set timeout API name to _set_default_timeout

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115197

Somehow the feedback does not show up, this PR is to address the comment in https://github.com/pytorch/pytorch/pull/115141.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @wz337 @tianyu-l @wconstab @yf225 @kiukchung @lucasllc

fduwjj committed:
[c10d] Change set timeout API name to _set_default_timeout

[ghstack-poisoned]

fduwjj commented on test/distributed/test_c10d_nccl.py:
```diff
@@ -1262,7 +1262,7 @@ def test_reset_nccl_pg_timeout(self):
         pg = dist.distributed_c10d._get_default_group()
         pg.allreduce(torch.rand(10).cuda(self.rank))
         self._check_nccl_timeout(timedelta(seconds=123))
-        pg._get_backend(torch.device(f"cuda:{self.rank}"))._reset_nccl_collective_timeout(23000)
```
directly passing in `23000` will lead to error:

```
 File "/data/users/fduwjj/pytorch/test/distributed/test_c10d_nccl.py", line 1265, in test_reset_nccl_pg_timeout
    pg._get_backend(torch.device(f"cuda:{self.rank}"))._set_default_timeout(23000)
TypeError: _set_default_timeout(): incompatible function arguments. The following argument types are supported:
    1. (self: torch._C._distributed_c10d.ProcessGroupNCCL, timeout_mil_sec: datetime.timedelta) -> None

Invoked with: <torch.distributed.distributed_c10d.ProcessGroupNCCL object at 0x7efd3e58b930>, 23000
```

cc: @H-Huang

wconstab replied:
i think this is ok. we use timedelta as our API for other dist. timeouts so its consistent.

fduwjj replied:
Yeah, this is just to mention that passing int directly does not work in this case. (Maybe float might work?)

XilunWu reviewed (approved):
lgtm!

fduwjj commented:
@pytorchbot merge

fduwjj commented:
@pytorchbot rebase

fduwjj committed:
Update on "[c10d] Change set timeout API name to _set_default_timeout"


Somehow the feedback does not show up, this PR is to address the comment in https://github.com/pytorch/pytorch/pull/115141.

cc H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab mrshenli zhaojuanmao rohan-varma kiukchung lucasllc XilunWu tianyu-l yf225

[ghstack-poisoned]

fduwjj commented:
@pytorchbot merge
